code/modules/mob/living/carbon/human/species_types/jellypeople.dm

```diff
@@ -479,88 +479,169 @@
+		action.Remove(old_body)
+	if(!QDELETED(new_body))
+		RegisterSignal(new_body, COMSIG_MOB_LOGIN, PROC_REF(login_notify))
+		RegisterSignal(new_body, COMSIG_MOB_DEATH, PROC_REF(link_death))
+		action.Grant(new_body)
+
+/datum/species/oozeling/stargazer/proc/link_death(mob/living/source_mob)
+	SIGNAL_HANDLER
+	if(!source_mob.mind || !(source_mob.mind in linked_minds))
+		return
+	var/datum/weakref/mind_ref = GLOB.slime_linked_with[source_mob.mind]
+	var/datum/species/oozeling/stargazer/stargazer = mind_ref?.resolve()
+	var/mob/living/carbon/human/link_owner = stargazer?.slimelink_owner?.resolve()
+	if(!link_owner || !istype(link_owner))
+		return
+	to_chat("<span class='danger'>As you die, you feel your link to [link_owner.real_name] fizzle out!</span>")
+	unlink_mind(source_mob.mind)
 
 /datum/species/oozeling/stargazer/on_species_gain(mob/living/carbon/C, datum/species/old_species)
 	..()
 	project_thought = new(src)
 	project_thought.Grant(C)
 	link_minds = new(src)
 	link_minds.Grant(C)
+	linked_speech = new(src)
+	linked_speech.Grant(C)
 	slimelink_owner = WEAKREF(C)
-	link_mob(C)
 
-/datum/species/oozeling/stargazer/proc/link_mob(mob/living/M)
-	if(QDELETED(M) || M.stat == DEAD)
+/datum/species/oozeling/stargazer/proc/link_mind(datum/mind/mind)
+	if(QDELETED(mind) || QDELETED(mind.current) || mind.current.stat == DEAD)
 		return FALSE
-	if(HAS_TRAIT(M, TRAIT_MINDSHIELD)) //mindshield implant, no dice
+	if(HAS_TRAIT(mind.current, TRAIT_MINDSHIELD)) //mindshield implant, no dice
 		return FALSE
 	var/mob/living/carbon/human/owner = slimelink_owner.resolve()
-	if(istype(M.get_item_by_slot(ITEM_SLOT_HEAD), /obj/item/clothing/head/foilhat))
+	if(istype(mind.current.get_item_by_slot(ITEM_SLOT_HEAD), /obj/item/clothing/head/foilhat))
 		if(owner)
-			to_chat(M, "<span class='danger'>[owner.real_name]'s no-good syndicate mind-slime is blocked by your protective headgear!</span>")
-
+			to_chat(mind.current, "<span class='danger'>[owner.real_name]'s no-good syndicate mind-slime is blocked by your protective headgear!</span>")
 		return FALSE
-	if(M in linked_mobs)
+	if(mind in linked_minds)
 		return FALSE
 	if(!owner)
 		return FALSE
-	linked_mobs.Add(M)
-	to_chat(M, "<span class='notice'>You are now connected to [owner.real_name]'s Slime Link.</span>")
+	linked_minds.Add(mind)
 	var/datum/action/innate/linked_speech/action = new(src)
 	linked_actions.Add(action)
-	action.Grant(M)
+	action.Grant(mind.current)
+	to_chat(mind.current, "<span class='notice'>You are now connected to [owner.real_name]'s Slime Link.</span>")
+	GLOB.slime_linked_with[mind] = WEAKREF(src)
+	RegisterSignal(mind.current, COMSIG_MOB_LOGIN, PROC_REF(login_notify))
+	RegisterSignal(mind.current, COMSIG_MOB_DEATH, PROC_REF(link_death))
+	RegisterSignal(mind, COMSIG_MIND_TRANSFER_TO, PROC_REF(mind_transfer))
+	to_chat(mind.current, "<span class='big notice'>You can use :[MODE_KEY_SLIMELINK] or .[MODE_KEY_SLIMELINK] to talk over the slime link!</span>")
```

Comment:
make this a proc or define since its copypastad a bunch

Reply:
Done